### PR TITLE
fix(cache): Invalidate cache of the compiler.

### PR DESCRIPTION
**Pull request by GerardPaligot**
Issue #575

### CODE REVIEW

File: src/test/java/spoon/test/api/APITest.java [hunk 1]
```diff
@@ -233,4 +235,30 @@ public void testPrintNotAllSourcesInCommandLine() throws Exception {
...
+		assertTrue(spoon.getModelBuilder().compile());
+
+		final CtClass<Bar> aClass = spoon.getFactory().Class().get(Bar.class);
+		aClass.addMethod(aMethod);
+		spoon.getModelBuilder().setShouldInvalidateCache(true);
+		assertFalse(spoon.getModelBuilder().compile());
+
+		aClass.removeMethod(aMethod);
+
+		spoon.getModelBuilder().setShouldInvalidateCache(true);
```
Comment (monperrus) [inline src/test/java/spoon/test/api/APITest.java]: why is it required to call twice `setShouldInvalidateCache(true)`?

**Comment by monperrus** [comment]
what if we don't have an option and we always invalidate the cache?

File: src/test/java/spoon/test/api/APITest.java (hunk 1 above)
Comment (GerardPaligot) [inline src/test/java/spoon/test/api/APITest.java]: A compiler is created at each build.

**Comment by GerardPaligot** [comment]
> what if we don't have an option and we always invalidate the cache?

Why have we a cache?

**Comment by tdurieux** [comment]
> Why have we a cache?
Because JDT asks 3 times for the content of the file during one compilation.

**Pull request merged**